### ISSUE TITLE
Make "Variant Tag" and "Dismiss variant" information configurable

### ISSUE THREAD

**Issue by mhkc** [Clinical-Genomics/scout#2561]
Our clinicians have historically used another set of terms for variant tag and dismiss variant in our fork of Scout.

We have discovered that our terms has IDs overlapping with the ones in the master repro of Scout, resulting in old samples having their classification altered to something nonsensical.

Therefore would it be nice if we could come up with a method of configuring these terms.

**Comment by dnil** [comment]
Agreed, config-by-constants is not exactly high-tech. The only redeeming feature about it is that there will be discussion between involved clinics and attempts to come up with good ones. I guess you see ours in constants now: could you list yours as well? Perhaps we could adopt some of them, and perhaps have some discussion at a joint user meeting?

**Comment by dnil** [comment]
(When I checked your GitHub fork some time ago, they looked very similar, so I guess you have a different copy somewhere?)

**Comment by mhkc** [comment]
We are using this file in production.

[variant_tags.prod.py.txt](https://github.com/Clinical-Genomics/scout/files/6363498/variant_tags.prod.py.txt)

While I think it would be a good idea to setup a user meeting to discuss the terminology, in my experience those are slow processes so it would be nice if we had a _official and blessed_ method of configuring the nomenclature. 

Also it might be worth considering hard coding the tags assigned to a variant in the database to avoid issues where a change in the CONSTANTS files impacts the results of previously analyzed samples as the changes might not reflect the intention by the analyst.

**Comment by dnil** [comment]
Ok. I agree it would be better to store the full classification record rather than just an int - that is the mongo philosophy: embed where possible. That or possibly use db entry classifications with unique ids, versioning etc.
It saves time, is exact - but wastes space and old cases do not benefit from updates fully. Let's do that!
We will likely need both a fix for the new cases (rather straightforward) in the code, as well as a db rewrite script to update the old ones (in itself easy, though always a little risky). Also, we could make these institute level config options, and give admins the authority to edit them (easy-intermediate). I like reaching a consensus, but these are numerous, and we could learn from each other anyway with like common presets and discussions.

I will say though that reusing the same ints (like PP -> VUS) was not terribly clever. There were effectively an infinite number of alternatives. I think someone realises that by now. 😆

The dismiss options are interesting; yours are very explicit and now-centric. This is not necessarily bad; I think we could rework a bit and reach a consensus if needed. As for keeping label and description identical, that has poped up locally here as well. I like it with different, but that is likely just because I have control over it. Let's change to at least have the label shown also on the report, and use descriptions mostly for tooltips or added explanations in the interface and report.

**Comment by mhkc** [comment]
>  We will likely need both a fix for the new cases (rather straightforward) in the code, as well as a db rewrite script to update the old ones (in itself easy, though always a little risky). Also, we could make these institute level config options, and give admins the authority to edit them (easy-intermediate). I like reaching a consensus, but these are numerous, and we could learn from each other anyway with like common presets and discussions.

Sounds good. I agree that we need a migration script that alters the database object. It would be great if we could configure the terms on an institute level, although some people will run wild with that.

> I will say though that reusing the same ints (like PP -> VUS) was not terribly clever. There were effectively an infinite number of alternatives. I think someone realises that by now. 😆

https://www.youtube.com/watch?v=sOVNVtX1ndU

> Let's change to at least have the label shown also on the report, and use descriptions mostly for tooltips or added explanations in the interface and report.

👍 

> The dismiss options are interesting; yours are very explicit and now-centric (common in gnomAD vs common in public databases). This is not necessarily bad; I think we could rework a bit and reach a consensus if needed.

I think it would be good to setup some kind of online workshop where the one or two representatives from Lund and Sthlm could discuss the terms and tags to find some compromise. I think it would be beneficial to use the same terms across Sweden, especially if more sites are going to be able to collaborate and share information. 

Who should  I contact to try setup such a meeting?
